test/compilable/stc_traits.d

```diff
@@ -0,0 +1,164 @@
+// REQUIRED_ARGS: -preview=dip1000 -preview=in
+/*
+TEST_OUTPUT:
+---
+100 tuple()
+101 tuple("return", "ref")
+102 tuple("ref")
+103 tuple()
+104 tuple("ref")
+105 tuple()
+106 tuple()
+107 tuple("ref")
+108 tuple("ref")
+109 tuple("ref")
+110 tuple("ref")
+111 tuple()
+112 tuple("ref")
+113 tuple("ref")
+114 tuple("ref")
+115 tuple("ref")
+116 tuple()
+117 tuple("ref")
+118 tuple("ref")
+119 tuple()
+120 tuple("ref")
+121 tuple("in")
+122 tuple("in")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("return", "ref")
+m-mixin tuple("return", "ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple()
+m-mixin tuple()
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("ref")
+m-mixin tuple("ref")
+m       tuple("in")
+m-mixin tuple("in")
+---
+*/
+
+void func(int i) {}
+void func(return ref bool i) {}
+void func(ref float a, int b) {}
+void get(T : int)(ref T t) {}
+void get()(float t) {}
+void get(T)(ref T[] t) {}
+void funcauto()(auto ref int i) {}
```

Comment:
maybe add an example with an `auto ref` parameter whose type is inferred.

Reply:
Added